x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_error.test.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { setMockValues } from '../../../__mocks__/kea_logic';
+
+import React from 'react';
+
+import { ApiError, Status } from '../../../../../common/types/api';
+
+import { ErrorStatePrompt } from '../../../shared/error_state';
+import { NotFoundPrompt } from '../../../shared/not_found';
+import { SendEnterpriseSearchTelemetry } from '../../../shared/telemetry';
+
+import { mountWithIntl } from '../../../test_helpers';
+
+import { EngineError } from './engine_error';
+
+describe('EngineError', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    setMockValues({});
+  });
+
+  it('renders 404 prompt for 404 error', () => {
```

Comment:
Do we want to handle permission denied status codes specially as well as 404s?

Reply:
That's a good idea, this was manually me handling edge case I knew of off hand. Probably worth an issue so we can do it in a unified way, and maybe in other places too. 

I don't love the error prompt used here since it just assumes you can't connect to `ent-search`.